src/graph.rs

```diff
@@ -1237,8 +1231,8 @@ impl<T> Graph<T>
     ///
     ///  assert!(graph.to_dot("example1", &mut f).is_ok());
     /// ```
-    pub fn to_dot(&self, graph_name: &str, output: &mut impl ::std::io::Write) -> Result<(), GraphErr> {
-        let edges : Vec<(_, _)> = self.edges.iter().map(|(w, _)| {
+    pub fn to_dot(&mut self, graph_name: &str, output: &mut impl ::std::io::Write) -> Result<(), GraphErr> {
+        let edges : Vec<(_, _)> = self.edges.keys().cloned().collect::<Vec<_>>().iter().map(|w| {
```

Comment:
This piece of code re-allocates all of the keys 2 times. Could you please revert the code to how it was before? If you insist on using `.keys()` it should be used entirely by itself like so:
```rust
self.edges.keys().map(|w| { ... })
```